include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,333 @@
+#define SOF_IPC4_GLB_MSG_TYPE(x)		((x) << SOF_IPC4_GLB_MSG_TYPE_SHIFT)
+
+/* pipeline creation ipc msg */
+#define SOF_IPC4_GLB_PIPE_INSTANCE_SHIFT	16
+#define SOF_IPC4_GLB_PIPE_INSTANCE_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GLB_PIPE_INSTANCE_ID(x)	((x) << SOF_IPC4_GLB_PIPE_INSTANCE_SHIFT)
+
+#define SOF_IPC4_GLB_PIPE_PRIORITY_SHIFT	11
+#define SOF_IPC4_GLB_PIPE_PRIORITY_MASK		GENMASK(15, 11)
+#define SOF_IPC4_GLB_PIPE_PRIORITY(x)		((x) << SOF_IPC4_GLB_PIPE_PRIORITY_SHIFT)
+
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT	0
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE_MASK		GENMASK(10, 0)
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE(x)		((x) << SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_EXT_LP_SHIFT		0
+#define SOF_IPC4_GL_PIPE_EXT_LP_MASK		BIT(0)
+#define SOF_IPC4_GL_PIPE_EXT_LP(x)		((x) << SOF_IPC4_GL_PIPE_EXT_LP_SHIFT)
+
+/* pipeline set state ipc msg */
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT	24
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_MASK	GENMASK(28, 24)
+#define SOF_IPC4_GL_PIPE_STATE_TYPE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_ID_SHIFT		16
+#define SOF_IPC4_GL_PIPE_STATE_ID_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GL_PIPE_STATE_ID(x)		((x) << SOF_IPC4_GL_PIPE_STATE_ID_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_SHIFT		0
+#define SOF_IPC4_GL_PIPE_STATE_MASK		GENMASK(15, 0)
+#define SOF_IPC4_GL_PIPE_STATE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_SHIFT)
+
+enum sof_ipc4_channel_config {
+	/* one channel only. */
+	SOF_IPC4_CHANNEL_CONFIG_MONO,
+	/* L & R. */
+	SOF_IPC4_CHANNEL_CONFIG_STEREO,
+	/* L, R & LFE; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_2_POINT_1,
+	/* L, C & R; MP3 & AAC only. */
+	SOF_IPC4_CHANNEL_CONFIG_3_POINT_0,
+	/* L, C, R & LFE; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_3_POINT_1,
+	/* L, R, Ls & Rs; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_QUATRO,
+	/* L, C, R & Cs; MP3 & AAC only. */
+	SOF_IPC4_CHANNEL_CONFIG_4_POINT_0,
+	/* L, C, R, Ls & Rs. */
+	SOF_IPC4_CHANNEL_CONFIG_5_POINT_0,
+	/* L, C, R, Ls, Rs & LFE. */
+	SOF_IPC4_CHANNEL_CONFIG_5_POINT_1,
+	/* one channel replicated in two. */
+	SOF_IPC4_CHANNEL_CONFIG_DUAL_MONO,
+	/* Stereo (L,R) in 4 slots, 1st stream: [ L, R, -, - ] */
+	SOF_IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_0,
+	/* Stereo (L,R) in 4 slots, 2nd stream: [ -, -, L, R ] */
+	SOF_IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_1,
+	/* L, C, R, Ls, Rs & LFE., LS, RS */
+	SOF_IPC4_CHANNEL_CONFIG_7_POINT_1,
+};
+
+enum sof_ipc4_interleaved_style {
+	SOF_IPC4_CHANNELS_INTERLEAVED,
+	SOF_IPC4_CHANNELS_NONINTERLEAVED,
+};
+
+enum sof_ipc4_sample_type {
+	SOF_IPC4_MSB_INTEGER, /* integer with Most Significant Byte first */
+	SOF_IPC4_LSB_INTEGER, /* integer with Least Significant Byte first */
+	SOF_IPC4_SIGNED_INTEGER, /* signed integer */
+	SOF_IPC4_UNSIGNED_INTEGER, /* unsigned integer */
+	SOF_IPC4_FLOAT, /* unsigned integer */
+};
+
+struct sof_ipc4_audio_format {
+	uint32_t sampling_frequency; /* sof_ipc4_sampling_frequency */
+	uint32_t bit_depth; /* sof_ipc4_bit_depth */
```

Comment:
what is `sof_ipc4_sampling_frequency` and `sof_ipc4_bit_depth`? Should these comment to be removed?

Reply:
> what is `sof_ipc4_sampling_frequency` and `sof_ipc4_bit_depth`? Should these comment to be removed?

That were the name of enums. Yes, they should be removed. Thanks.